test/infcover.c

```diff
@@ -590,7 +595,9 @@ static void cover_inflate(void) {
     try("4 80 49 92 24 49 92 24 f b4 ff ff c3 84", "invalid distances set", 1);
     try("4 c0 81 8 0 0 0 0 20 7f eb b 0 0", "invalid literal/length code", 1);
     try("2 7e ff ff", "invalid distance code", 1);
+#ifdef INFLATE_ALLOW_INVALID_DISTANCE_TOOFAR_ARRR
     try("c c0 81 0 0 0 0 0 90 ff 6b 4 0", "invalid distance too far back", 1);
+#endif
```

Comment:
This is the part I was wondering about earlier, and I am again now. So it'll probably need a comment if it is correct.
This looks to me like we disable the check of invalid distance unless we allow those invalid distances.

Isn't this supposed to check that we do return the correct error when distance is too far back? In that case, we should probably be testing for this by default, unless we allow the invalid distances, then it makes little sense to check them.

Reply:
We disable the check because that logical branch is not supported. So, when `INFLATE_ALLOW_INVALID_DISTANCE_TOOFAR_ARRR` is defined, it will allow invalid distances. The try() function simply adds code coverage to the logic that is wrapped in `#ifdef INFLATE_ALLOW_INVALID_DISTANCE_TOOFAR_ARRR`.

Reply:
Also, if allowing invalid distances in zlib-ng is something we want to do let me know. We can't just do it for infcover only because codecov will not see a zlibstatic-with-invalid-dist as the same thing as zlibstatic. So I left it turned off as is. And only enable check if turned on.

Reply:
I thought these tests are testing invalid behavior? If so, then shouldn't the test be enabled if too far distances are not allowed?
I have never completely understood infcover though, too much code, too little comments.

Reply:
Are you suggesting we just get rid of the #define and always allow invalid distances in the lib?

Reply:
Not at all.
I suspect we are talking past each other.

Reply:
It does test invalid behavior. I don’t know a clean way to enable it. We can’t just enable it for this application because setting the #define would apply to all apps.
If we have a separate library with only the #define then I found that code coverage doesn’t see the zlib as the same.

Reply:
I mean, isn't it supposed to test the invalid behavior and isn't it supposed to fail (and be catched by infcover, returning success if it failed) when the define isn't set? Doesn't this test check that the error is returned if it is invalid? Aren't we testing error-handling?
If we only run the test when it is not in fact invalid, then it would be a pointless test..  Do you see what I mean?

Reply:
I made some modifications so that code still gets run even when the `#define` is not set.

Reply:
I made the wrong assumption that the code did not need to be called if `#define` was not set. However, it should still be called to run the failure code path.

Reply:
Looks good now :)